File: go/custom_property_definition_update_request.go
// this file is @generated
package meteroid

import "encoding/json"

// Update of a definition. `key`, `entity_type` and `property_type` are immutable and cannot be
// changed here. Any field left absent is unchanged.
type CustomPropertyDefinitionUpdateRequest struct {
	Config *PropertyConfig `json:"config,omitempty"`

	DefaultValue json.RawMessage `json:"default_value,omitempty"`

	Description *string `json:"description,omitempty"`

	DisplayOrder *int32 `json:"display_order,omitempty"`

	Name *string `json:"name,omitempty"`

	Required *bool `json:"required,omitempty"`
}

File: python/meteroid/models/custom_property_definition_update_request.py
# this file is @generated
import dataclasses
import typing as t

from ..serialization import BaseModel
from .property_config import PropertyConfig


@dataclasses.dataclass
class CustomPropertyDefinitionUpdateRequest(BaseModel):
    """Update of a definition. `key`, `entity_type` and `property_type` are immutable and cannot be
    changed here. Any field left absent is unchanged."""

    config: t.Optional[PropertyConfig] = None

    default_value: t.Optional[t.Any] = None

    description: t.Optional[str] = None

    display_order: t.Optional[int] = None

    name: t.Optional[str] = None

    required: t.Optional[bool] = None
